nixos/modules/system/boot/systemd/initrd.nix

```diff
@@ -493,7 +493,7 @@ in {
 
           # If we are not booting a NixOS closure (e.g. init=/bin/sh),
           # we don't know what root to prepare so we don't do anything
-          if ! [ -x "/sysroot$closure/prepare-root" ]; then
+          if ! [ -x "/sysroot$(readlink "/sysroot$closure/prepare-root" || echo "$closure/prepare-root")" ]; then
```

Comment:
Can do `readlink -f` here to avoid the `|| echo`.

Reply:
That completely changes the behavior compared to what I have used:

`readlink -f /sysroot/[symlink]` -> `/[real]`
`readlink -f /sysroot/[real]` -> `/sr/[real]`

We would need to know if it's a symlink or not and adjust the way the path is used accordingly, which would basically amount to using `readlink ||`, which I used a simpler version of.

Reply:
Oh heck. My bad.

Reply:
Believe me that was the first thing I tried, I also had no luck with `realpath` (`realpath --relative-base=/sysroot /sysroot$closure` gets us close but may or may not start with a `/`, `realpath $(realpath --relative-base=/sysroot /sysroot/$closure)` may have worked, but I liked that less). It would have been nice to just use a command and not need any logic alongside it.